### PR TITLE
Fix Celery task executing in multiple instances

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights

This Pull Request encompasses two significant changes aimed at improving task management and enhancing logging capabilities within our set of EC2 instances.

- Improved One-off Task Execution: We encountered an issue where tasks marked as 'one-off' were executing across all 5 EC2 instances, often leading to redundancy. This PR addresses this issue, optimizing the execution of 'one-off' tasks to ensure they only run if they haven't run previously. This strategy results in a much more efficient distribution of tasks across our instances and reduces unnecessary redundancy.

- Revamped Logging System: In parallel with the above fix, we decided to improve our monitoring and logging capabilities. Initially, our logging method revolved around using a Slack webhook. However, to enhance efficiency, we decided to transition into using CloudWatch for logging. This change not only strengthens our tracking and debugging capabilities but also allows us to monitor our systems with an added layer of precision

#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [x] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

File: src/task_queue/tasks.py [hunk 1]
```diff
@@ -1,63 +1,62 @@
...
+					log_status(task.status, task.job_name, "Task ran successfully" if res else "Task failed")
+				
+				except Exception as e:
+					task.status = "FAILED"
+					log_status(task.status, task.job_name, today, str(e))
+			else:
+				task.status = "FAILED"
+				log_status(task.status, task.job_name, "Task not found in FUNCTIONS")
+			task.save()
+			if task.frequency == "ONE_OFF":
```
Comment (BradHN1) [inline src/task_queue/tasks.py:59]: PLEASE - don't archive ONE_OFF tasks.  Leave them in the list so they can run again.

Comment (abdullai-t) [reply src/task_queue/tasks.py:59]: Ok

**Review by archx3 — APPROVED** [review]

File: src/task_queue/tasks.py [hunk 2]
```diff
@@ -1,63 +1,61 @@
...
+from .models import Task
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+SUCCESS_LOG_KEY = "CELERY_TASK_SUCCESS"
+FAILURE_LOG_KEY = "CELERY_TASK_FAILURE"
+
+
+def log_status(status, task_name, message):
```
Comment (Opoku-Agyemang) [inline src/task_queue/tasks.py:19]: maybe we can send the slack message here

**Review by Opoku-Agyemang — APPROVED** [review]

**Review by Opoku-Agyemang — APPROVED** [review]
lgtm!

**Pull request merged**